### PR TITLE
Tail Entwining

### PR DESCRIPTION
**Pull request by Geevies**
* Ported tail entwining. Simply grab another person that has a tail to entwine tails with them. Make sure to do it in the dorm or risk a ban!

Ported from: https://github.com/tgstation/tgstation/pull/54991

A feedback thread won't be necessary, we already have permission from the shadow cabal.
image

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by optimumtact** [comment]
Fixes critical security issue too

**Comment by NonQueueingMatt** [comment]
The fact that I considered this a serious PR because you made it until I opened it is just bruh

**Review by MarinaGryphon — APPROVED** [review]
Should probably use `SPAN_DANGER` but ehh.

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 1]
```diff
@@ -199,7 +199,10 @@
 				G.last_action = world.time
 				visible_message("<span class='warning'>[M] gets a strong grip on [src]!</span>")
 				return 1
-			visible_message("<span class='warning'>[M] has grabbed [src] passively!</span>")
+			if(src.has_tail() && H.has_tail())
+				H.visible_message(SPAN_DANGER("[H] coils their tail with [src], wow is that okay in public?!"), "You entwine your tail with [src]'s!")
```
Comment (Doxxmedearly) [inline code/modules/mob/living/carbon/human/human_attackhand.dm:203]:
```suggestion
				H.visible_message(SPAN_DANGER("[H] coils their tail with [src], wow is that okay in public?!"), "You entwine your tail with [src]'s!")
				qdel(H)
```

Comment (Geevies) [reply code/modules/mob/living/carbon/human/human_attackhand.dm:203]: i can raise you one better

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 2]
```diff
@@ -199,7 +199,10 @@
 				G.last_action = world.time
 				visible_message("<span class='warning'>[M] gets a strong grip on [src]!</span>")
 				return 1
-			visible_message("<span class='warning'>[M] has grabbed [src] passively!</span>")
+			if(src.has_tail() && H.has_tail())
+				H.visible_message(SPAN_WARNING("[H] coils their tail with [src], wow is that okay in public?!"), "You entwine your tail with [src]'s!")
```
Comment (MarinaGryphon) [inline code/modules/mob/living/carbon/human/human_attackhand.dm]:
```suggestion
				var/in_public = length(viewers(H)+viewers(src)) > 3
				H.visible_message(span(in_public ? "danger" : "warning, "[H] coils their tail with [src]'." + (in_public) ? "Wow, is that okay in public?!" : ""), SPAN_NOTICE("You entwine your tail with [src]'s!"))
```

**Review by MarinaGryphon — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/human_attackhand.dm (hunk 2 above)
Comment (Geevies) [reply code/modules/mob/living/carbon/human/human_attackhand.dm]: it's outdated i can't merge it but @oranges you should backport this

Comment (Geevies) [reply code/modules/mob/living/carbon/human/human_attackhand.dm]: oh fuck it pinged a random oranges

Comment (MarinaGryphon) [reply code/modules/mob/living/carbon/human/human_attackhand.dm]: @optimumtact high priority fix

Comment (optimumtact) [reply code/modules/mob/living/carbon/human/human_attackhand.dm]: thanks I'll use this

**Comment by Geevies** [comment]
ran into some issues with my git client, gonna reopen instead of force pushing to preserve history

**Comment by NonQueueingMatt** [comment]
dont open

**Pull request closed without merging**